src/vs/workbench/contrib/terminal/browser/links/terminalLinkQuickpick.ts

```diff
@@ -0,0 +1,101 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { EventType } from 'vs/base/browser/dom';
+import { localize } from 'vs/nls';
+import { IClipboardService } from 'vs/platform/clipboard/common/clipboardService';
+import { IPickOptions, IQuickInputService, IQuickPickItem } from 'vs/platform/quickinput/common/quickInput';
+import { IDetectedLinks } from 'vs/workbench/contrib/terminal/browser/links/terminalLinkManager';
+import { TerminalLinkQuickPickEvent } from 'vs/workbench/contrib/terminal/browser/terminal';
+import { ILink } from 'xterm';
+
+export class TerminalLinkQuickpick {
+	constructor(
+		@IQuickInputService private readonly _quickInputService: IQuickInputService,
+		@IClipboardService private readonly _clipboardService: IClipboardService
+	) { }
+
+	async show(links: IDetectedLinks): Promise<void> {
+		const wordPicks = links.wordLinks ? await this._generatePicks(links.wordLinks) : undefined;
+		const filePicks = links.fileLinks ? await this._generatePicks(links.fileLinks) : undefined;
+		const webPicks = links.webLinks ? await this._generatePicks(links.webLinks) : undefined;
+
+		const picks: IQuickPickItem[] = [];
+		if (wordPicks) {
+			picks.push({ label: localize('terminal.integrated.wordLinks', "Word") });
+		}
+		if (filePicks) {
+			picks.push({ label: localize('terminal.integrated.fileLinks', "File") });
+		}
+		if (webPicks) {
+			picks.push({ label: localize('terminal.integrated.webLinks', "Web") });
+		}
```

Comment:
Perhaps I am misunderstanding what we're looking for here. 

Imo, broadening the scope for what will already be a long list of options won't be helpful from an accessibility perspective.

Reply:
They are already ordered by reverse line within the viewport. 

Let's seek feedback and also test this using a screen reader

Reply:
I think it might look pretty bad to have those as a description. i'll try it though. since you're suggesting they should all be opened the same way (instead of copying the word), seems fine if they're intermixed since they function the same way